Cargo.toml

```diff
@@ -0,0 +1,9 @@
+[workspace]
+members = [
```

Comment:
This will conflict with the workspace code you added to `Compile.hs` earlier.  We actually used to have this section here some time ago, but ended up removing it.  I believe the reason was that `cargo` complained when adding a dependency on the generated Rust project from external crates, as it does not like seeing workspaces anywhere other than in the main crate.

Also, this breaks CI build.

Reply:
It shouldn't conflict with the `Compile.hs` code since generated code should be out-of-tree, I may be able to fix the test errors by adding `exclude = ["test/*"]`

Reply:
Ok, let's work on the workspace thing then, but I think it can be a separate PR.  I like the other things you did here and would like to merge, but I feel like doing workspaces properly may take some more time.

Reply:
> It shouldn't conflict with the `Compile.hs` code since generated code should be out-of-tree except in the case of tests, and I may be able to fix those errors by adding `exclude = ["test/*"]` to the root `Cargo.toml`.

Yes, but the generated code will then contain two workspace sections, won't it?

Reply:
It won't since this `Cargo.toml` only affects the source directory of `differential-datalog` itself, while the generated code (generally elsewhere) has its own, separate workspace

Reply:
My bad, I only now realized this is a separate `.toml`. In that case I have no objections as long as we can fix the CI.

Reply:
You were indeed correct and it was trickier than I anticipated, so I'll do it separately